unity-renderer/Assets/DCLPlugins/EquippedEmotes/EquippedEmotesPlugin.cs

```diff
@@ -0,0 +1,106 @@
+using DCL.Helpers;
+using Newtonsoft.Json;
+using System.Collections.Generic;
+
+namespace DCL.EquippedEmotes
+{
+    /// <summary>
+    /// Plugin feature that initialize the Equipped Emotes feature.
+    /// </summary>
+    public class EquippedEmotesPlugin : IPlugin
+    {
+        internal const string PLAYER_PREFS_EQUIPPED_EMOTES_KEY = "EquippedNFTEmotes";
+
+        internal DataStore_EmotesCustomization emotesCustomizationDataStore => DataStore.i.emotesCustomization;
+
+        public EquippedEmotesPlugin()
+        {
+            LoadDefaultEquippedEmotes();
+
+            emotesCustomizationDataStore.isEmotesCustomizationInitialized.OnChange += OnEmotesCustomizationInitialized;
+            OnEmotesCustomizationInitialized(emotesCustomizationDataStore.isEmotesCustomizationInitialized.Get(), false);
+
+            emotesCustomizationDataStore.equippedEmotes.OnSet += SaveEquippedEmotesInLocalStorage;
+        }
+
+        private void OnEmotesCustomizationInitialized(bool current, bool previous)
+        {
+            if (!current)
+                return;
+
+            emotesCustomizationDataStore.isEmotesCustomizationInitialized.OnChange -= OnEmotesCustomizationInitialized;
+            LoadEquippedEmotesFromLocalStorage();
+        }
+
+        internal void LoadDefaultEquippedEmotes() { SetEquippedEmotes(GetDefaultEmotes()); }
+
+        internal void LoadEquippedEmotesFromLocalStorage()
+        {
+            List<string> storedEquippedEmotes;
+
+            try
+            {
+                storedEquippedEmotes = JsonConvert.DeserializeObject<List<string>>(PlayerPrefsUtils.GetString(PLAYER_PREFS_EQUIPPED_EMOTES_KEY));
+            }
+            catch
+            {
+                storedEquippedEmotes = null;
```

Comment:
I guess we can just set `storedEquippedEmotes = GetDefaultEmotes()`

Reply:
This try/catch is taking into account 2 possible null results:
1. If there is no any `EquippedNFTEmotes` key stored in PlayerPrefs.
2. If there is an `EquippedNFTEmotes` stored in PlayerPrefs BUT the data model doesn't match with the implemented one (it shouldn't never happen but it is as an extra security check).

```
try
{
    storedEquippedEmotes = JsonConvert.DeserializeObject<List<string>>(PlayerPrefsUtils.GetString(PLAYER_PREFS_EQUIPPED_EMOTES_KEY));
}
catch
{
    storedEquippedEmotes = null;
}
```

So I'm doing the `GetDefaultEmotes()` assignation out of the try/catch in order to do write the call only one time, whatever the case happened.

```
if (storedEquippedEmotes == null)
    storedEquippedEmotes = GetDefaultEmotes();
```

I thought it was cleaner to do it this way.

Reply:
You are right, we could have done it in a `finally` clause, but it's fine this way.